relay-server/src/endpoints/events.rs

```diff
@@ -28,9 +29,73 @@ fn get_captured_event(event_id: Path<EventId>) -> ResponseFuture<HttpResponse, M
     Box::new(future)
 }
 
+#[derive(Debug, Serialize)]
+struct JsonEnvelope {
+    headers: EnvelopeHeaders,
+    items: Vec<JsonItem>,
+}
+
+#[derive(Debug, Serialize)]
+struct JsonItem {
+    headers: ItemHeaders,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    payload: Option<serde_json::Value>,
+}
+
+fn envelope_as_json(envelope: Result<Envelope, String>) -> Result<JsonEnvelope, String> {
+    envelope.map(|envelope| {
+        let items = envelope
+            .items()
+            .map(|item| {
+                let payload = if let Some(ContentType::Json) = item.content_type() {
+                    Some(serde_json::from_slice(item.payload().as_ref()).unwrap())
+                } else {
+                    None
+                };
+
+                JsonItem {
+                    headers: item.headers().clone(),
+                    payload,
+                }
+            })
+            .collect::<Vec<_>>();
+
+        JsonEnvelope {
+            headers: envelope.headers().clone(),
+            items,
+        }
+    })
+}
+
+fn get_last_event(_: ()) -> ResponseFuture<HttpResponse, MailboxError> {
+    let future = EnvelopeManager::from_registry()
+        .send(GetCapturedEnvelope { event_id: None })
+        .map(|result| match result {
+            Some(result) => {
+                let envelope = envelope_as_json(result);
+
+                match serde_json::to_string(&envelope) {
+                    Ok(json) => HttpResponse::Ok()
+                        .content_type("application/json")
+                        .body(json),
+                    Err(e) => HttpResponse::BadGateway()
+                        .content_type("text/plain")
+                        .body(e.to_string()),
+                }
+            }
+            None => HttpResponse::NotFound().finish(),
+        });
+
+    Box::new(future)
+}
+
 pub fn configure_app(app: ServiceApp) -> ServiceApp {
     app.resource("/api/relay/events/{event_id}/", |r| {
         r.name("internal-events");
         r.method(Method::GET).with(get_captured_event);
     })
+    .resource("/api/relay/last-event/", |r| {
```

Comment:
I'd call this something like `last-request` so we can also return more information about the latest request. Specifically, if we drop an envelope, we'll likely want to retain the outcome and return it here.